app/(playground)/p/[agentId]/contexts/execution.tsx

```diff
@@ -578,20 +589,27 @@ export function ExecutionProvider({
 			});
 
 			const { blobUrl } = await putExecutionAction(executionSnapshot);
+			const runEndedAt = Date.now();
 			dispatch({
 				type: "addExecutionIndex",
 				input: {
 					executionIndex: {
 						executionId: currentExecution.id,
 						blobUrl,
-						completedAt: Date.now(),
+						completedAt: runEndedAt,
 					},
 				},
 			});
 
+			await recordAgentUsageAction(
+				currentExecution.runStartedAt,
+				runEndedAt,
+				currentExecution.durationMs,
+			);
+
```

Comment:
I would like to separate the recording of the execution from the context of the execution, so it would be better to make it a callback like onFinishPerformFlowExecution and add props like onFinishPerformFlowExecutionAction in the Provider. I think, but in this Pull request, it is fine as it is.